src/main/java/com/puppycrawl/tools/checkstyle/xpath/AbstractNode.java

```diff
@@ -145,7 +145,14 @@ public AxisIterator iterateAxis(byte axisNumber, NodeTest nodeTest) {
      */
     @Override
     public int compareOrder(NodeInfo nodeInfo) {
```

Comment:
We need to update javadoc to explain a reason of such custom ordering.

Reply:
done